### DISCUSSION TITLE
Is there a way to assign a sheet to Wave Port

### DISCUSSION THREAD

**Discussion by anoushkanarayan** [ansys/pyaedt#3029] [Q&A] [answered]
I have this sheet which is currently unassigned. My solution type is already set to Modal, so that is not the issue. I am unable to find a method that will help me assign this from "Unassigned" to "Wave Port."

sheet1 = hfss.modeler.primitives.create_rectangle(csPlane="YZ", position=[0, "-p_w/2", "-p_h/2+dz"], dimension_list=["p_w", "p_h"], name="p1")

Thank you!

**Comment by MaxJPRey** [comment, marked as answer]
Hi @anoushkanarayan ,
Did you try something like this:

```
port = self.aedtapp.wave_port(
    signal=sheet1,
    deembed=5,
    integration_line=self.aedtapp.AxisDir.XNeg,
    impedance=40,
    num_modes=2,
    name="sheet1_Port",
    renormalize=False,
    reference=["outer"],
    terminals_rename=False,
)
```

It should automatically set the sheet to "Wave Port" as soon as the port is created and assigned to the sheet.
